core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -1089,17 +1091,15 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
   private def initHadoopOutputMetrics(context: TaskContext): (OutputMetrics, Option[() => Long]) = {
     val bytesWrittenCallback = SparkHadoopUtil.get.getFSBytesWrittenOnThreadCallback()
     val outputMetrics = new OutputMetrics(DataWriteMethod.Hadoop)
-    if (bytesWrittenCallback.isDefined) {
-      context.taskMetrics.outputMetrics = Some(outputMetrics)
```

Comment:
Yeah, good point. It isn't really straightforward to add outputmetrics for older hadoop. I'll revert this change so that outputMetrics are only created for hadoop > 2.5. We can add a JIRA for adding output metrics for older hadoop.